browser/brave_vpn/dns/brave_vpn_dns_observer_service.cc

```diff
@@ -0,0 +1,141 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_vpn/dns/brave_vpn_dns_observer_service.h"
+
+#include <vector>
+
+#include "base/bind.h"
+#include "base/strings/string_split.h"
+#include "base/strings/string_util.h"
+#include "brave/browser/ui/views/brave_vpn/vpn_notification_dialog_view.h"
+#include "brave/components/brave_vpn/brave_vpn_utils.h"
+#include "brave/components/brave_vpn/pref_names.h"
+#include "chrome/browser/net/secure_dns_config.h"
+#include "chrome/browser/net/secure_dns_util.h"
+#include "chrome/browser/net/stub_resolver_config_reader.h"
+#include "chrome/browser/net/system_network_context_manager.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/browser/ui/browser_window.h"
+#include "chrome/browser/ui/simple_message_box.h"
+#include "chrome/common/pref_names.h"
+#include "chrome/grit/chromium_strings.h"
+#include "components/grit/brave_components_strings.h"
+#include "components/policy/policy_constants.h"
+#include "components/prefs/pref_service.h"
+#include "net/dns/dns_config.h"
+#include "net/dns/public/dns_over_https_config.h"
+#include "net/dns/public/doh_provider_entry.h"
+#include "net/dns/public/secure_dns_mode.h"
+#include "ui/base/l10n/l10n_util.h"
+
+namespace brave_vpn {
+
+namespace {
+const char kCloudflareDnsProviderURL[] =
+    "https://chrome.cloudflare-dns.com/dns-query";
+
+void SkipDNSDialog(PrefService* prefs, bool checked) {
+  if (!prefs)
+    return;
+  prefs->SetBoolean(prefs::kBraveVPNShowDNSPolicyWarningDialog, !checked);
+}
+
+std::string GetDoHServers(const std::string* user_servers) {
+  return user_servers && !user_servers->empty() ? *user_servers
```

Comment:
we don't care about this here. We only care if vpn is enabled or not

Reply:
this function was removed